### PR TITLE
[SYCL][UnitTest] Add unittest for `sycl_ext_oneapi_virtual_mem` extension

### PR DESCRIPTION
**Pull request by DYNIO-INTEL**
Based on the test plan https://github.com/intel/llvm/pull/15509, this PR adds an unit-test checking if extension throws expected exception when one of the devices does not have `sycl::aspect::ext_oneapi_virtual_mem`.

### CODE REVIEW

File: sycl/unittests/Extensions/VirtualMemory/Exceptions.cpp [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+  auto params = *static_cast<ur_device_get_params_t *>(pParams);
+  if (*params.ppNumDevices)
+    **params.ppNumDevices = 3;
+  if (*params.pphDevices) {
+    if (*params.pNumEntries > 0)
+      (*params.pphDevices)[0] = GlobalDevicesHandle[0];
+    if (*params.pNumEntries > 1)
+      (*params.pphDevices)[1] = GlobalDevicesHandle[1];
+    if (*params.pNumEntries > 2)
+      (*params.pphDevices)[2] = GlobalDevicesHandle[2];
```
Comment (aelovikov-intel) [inline sycl/unittests/Extensions/VirtualMemory/Exceptions.cpp:34]: Why not a simple loop?

File: sycl/unittests/Extensions/VirtualMemory/Exceptions.cpp [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+  }
+  return UR_RESULT_SUCCESS;
+}
+
+template <bool VirtualMemSupported>
+ur_result_t after_urDeviceGetInfo_AllDevices(void *pParams) {
+  auto params = reinterpret_cast<ur_device_get_info_params_t *>(pParams);
+  switch (*params->ppropName) {
+  case UR_DEVICE_INFO_VIRTUAL_MEMORY_SUPPORT: {
```
Comment (aelovikov-intel) [inline sycl/unittests/Extensions/VirtualMemory/Exceptions.cpp:43]: Can we just have early returns instead of switches everywhere?

Comment (DYNIO-INTEL) [reply sycl/unittests/Extensions/VirtualMemory/Exceptions.cpp:43]: Sure, can be done

**Review by aelovikov-intel — APPROVED** [review]

**Comment by DYNIO-INTEL** [comment]
@intel/llvm-gatekeepers Can we merge this PR?

**Pull request merged**